kornia/geometry/transform/pyramid.py

```diff
@@ -224,3 +225,46 @@ def pyrup(input: torch.Tensor, border_type: str = 'reflect') -> torch.Tensor:
     See :class:`~kornia.transform.PyrUp` for details.
     """
     return PyrUp(border_type)(input)
+
+
+def build_pyramid(
+        input: torch.Tensor,
+        max_level: int,
+        border_type: str = 'reflect') -> List[torch.Tensor]:
+    r"""Constructs the Gaussian pyramid for an image.
+
+    The function constructs a vector of images and builds the Gaussian pyramid
+    by recursively applying pyrDown to the previously built pyramid layers.
+
+    Args:
+        input (torch.Tensor): the tensor to be used to constructuct the pyramid.
+        max_level (int): 0-based index of the last (the smallest) pyramid layer.
+          It must be non-negative.
+        borde_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``, ``'reflect'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'reflect'``.
+    Shape:
+        - Input: :math:`(B, C, H, W)`
+        - Output :math:`[(B, NL, C, H, W), (B, NL, C, H/2, W/2), ...]`
+    """
+    if not isinstance(input, torch.Tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not len(input.shape) == 4:
+        raise ValueError(f"Invalid input shape, we expect BxCxHxW. Got: {input.shape}")
+
+    if not isinstance(max_level, int) or max_level < 0:
+        raise ValueError(f"Invalid max_level, it must be a positive integer. Got: {max_level}")
+
+    # create empty list and append the original image
+    pyramid: List[torch.Tensor] = []
+    pyramid.append(input)
+
+    # iterate and downsample
+
+    for _ in range(max_level - 1):
+        img_curr: torch.Tensor = pyramid[-1]
```

Comment:
@ducha-aiki should we apply `F.avg_pool2d` similar to https://github.com/arraiyopensource/kornia/pull/216/files#diff-ef83ccf8ba25889bf4a53b6ad45518b6R186 ?

Reply:
I don`t understand the question. We indeed apply it in pyrdown function, which is called below
[-1] refers to the last element of the list

Reply:
yeap, then I don't understand why you use `avg_pool2d`  in here https://github.com/arraiyopensource/kornia/blob/master/kornia/geometry/transform/pyramid.py#L185-L186

Reply:
Because pyrdown is (fixed) blur + 2x downsampling.
In the "scale pyramid" I need to be able to apply variable blur, depending on how many levels inside one spatial resolution I need to have. 
This two functions are of similar nature, but different use-cases. Same as in OpenCV

Reply:
ok I see. Then I'll leave like this